ddl/util/util.go

```diff
@@ -325,3 +326,41 @@ func WrapKey2String(key []byte) string {
 	}
 	return fmt.Sprintf("0x%x", key)
 }
+
+const (
+	getRaftKvVersionSQL = "show config where type = 'tikv' && name = 'storage.engine'"
+	raftKv2             = "raft-kv2"
+)
+
+// IsRaftKv2 checks whether the raft-kv2 is enabled
+func IsRaftKv2(ctx context.Context, sctx sessionctx.Context) (bool, error) {
+	// Mock store does not support `show config` now, so we  use failpoint here
+	// to control whether we are in raft-kv2
+	failpoint.Inject("isRaftKv2", func(v failpoint.Value) (bool, error) {
+		v2, _ := v.(bool)
+		return v2, nil
+	})
+
+	rs, err := sctx.(sqlexec.SQLExecutor).ExecuteInternal(ctx, getRaftKvVersionSQL)
+	if rs != nil {
+		defer terror.Call(rs.Close)
+	}
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	req := rs.NewChunk(nil)
+	it := chunk.NewIterator4Chunk(req)
+	err = rs.Next(context.TODO(), req)
```

Comment:
we can use the context in argument

Reply:
ditto

Reply:
[3806d90](https://github.com/pingcap/tidb/pull/45482/commits/3806d90b9350d6248f375a39a60f4b9d85e80d4c)